include/unifex/when_all.hpp

```diff
@@ -213,19 +213,39 @@ struct _op<Receiver, Senders...>::type {
 
   void start() noexcept {
     stopCallback_.construct(
-        get_stop_token(receiver_), cancel_operation{stopSource_});
+        get_stop_token(receiver_),
+        cancel_operation<Receiver, Senders...>{*this});
     ops_.start();
   }
 
-  private:
+  void request_stop() noexcept {
+    // mark callback as running (own deliver_result)
+    auto oldCount = refCount_.fetch_add(1, std::memory_order_relaxed);
+    UNIFEX_ASSERT(!(oldCount & callback_running_bit));
+    stopSource_.request_stop();
+    // unmark (deliver_result back to operation)
+    oldCount = refCount_.fetch_sub(1, std::memory_order_relaxed);
```

Comment:
I think this at least needs to be a `acq_rel` so that the `set_done()` call below has visibility of any other operations that completed while we were calling `request_stop()`.

It needs to be `release` to ensure that the store to the stopSource_ in `request_stop()` is visible to another thread that sees our write and goes on to destroy the operation state.